accounting/aggregations/query.py

```diff
@@ -0,0 +1,547 @@
+    parser.add_argument("--end", type=valid_date, default=datetime.strftime(datetime.now(), "%Y-%m-%d"),
+                        help="the date to end reporting on 'YYYY-MM-DD', defaults to the current date")
+
+    es_opts = parser.add_argument_group("elasticsearch options")
+    for name, props in ELASTICSEARCH_ARGS.items():
+        es_opts.add_argument(name, **props)
+
+    output_opts = parser.add_argument_group("email options")
+    for name, props in OUTPUT_ARGS.items():
+        output_opts.add_argument(name, **props)
+
+    return parser.parse_args()
+
+
+# percentages for the totals row is calculated in python
+# due to limitations with calculating percents in ES
+def calc_totals_percents(resp) -> dict:
+    ret = {}
+    buckets = resp.aggregations.to_dict()
+
+    # TODO what to do about the previously stored pretty names?
+    ret.update({"% Goodput" : 
+                buckets["good_core_hours"]["value"] / (buckets["cpu_core_hours"]["value"] if buckets["cpu_core_hours"]["value"] else 0) * 100}) 
+    ret.update({"% Ckptable" : 
+                buckets["ckptable_filt"]["doc_count"] / (buckets["uniq_job_ids"]["value"] if buckets["uniq_job_ids"]["value"] else 0) * 100})
+    ret.update({"% Removed" : 
+                buckets["rmd_filt"]["doc_count"] / (buckets["cpu_core_hours"]["value"] if buckets["cpu_core_hours"]["value"] else 0) * 100})
+    ret.update({"Shadow Starts / ID" : 
+                buckets["num_shadw_starts"]["value"] / (buckets["uniq_job_ids"]["value"] if buckets["uniq_job_ids"]["value"] else 0)})
+    ret.update({"Exec Att / Shadow Start" : 
+                buckets["num_exec_attempts"]["value"] / (buckets["num_shadw_starts"]["value"] if buckets["num_shadw_starts"]["value"] else 0)})
+    ret.update({"Holds / ID" : 
+                buckets["num_holds"]["value"] / (buckets["uniq_job_ids"]["value"] if buckets["uniq_job_ids"]["value"] else 0)})
+    ret.update({"% Short" : 
+                buckets["short_jobs"]["doc_count"] / (buckets["uniq_job_ids"]["value"] if buckets["uniq_job_ids"]["value"] else 0) * 100})
+    ret.update({"% Restarted" : 
+                buckets["restarted_jobs"]["doc_count"] / (buckets["uniq_job_ids"]["value"] if buckets["uniq_job_ids"]["value"] else 0) * 100})
+    ret.update({"% Held" : 
+                buckets["held_jobs"]["doc_count"] / (buckets["uniq_job_ids"]["value"] if buckets["uniq_job_ids"]["value"] else 0) * 100})
+    ret.update({"% Over Req. Disk" : 
+                buckets["over_disk_jobs"]["doc_count"] / (buckets["uniq_job_ids"]["value"] if buckets["uniq_job_ids"]["value"] else 0) * 100})
+    ret.update({"% S'ty Jobs" : 
+                buckets["sty_jobs"]["doc_count"] / (buckets["uniq_job_ids"]["value"] if buckets["uniq_job_ids"]["value"] else 0) * 100})
+
+    return ret
+
+# =========== end of helper functions ===========
+
+def main():
+    # parse arguments
+    args = parse_args()
+
+    client = elasticsearch.Elasticsearch(hosts=[args.host], timeout=120)
```

Comment:
We'll need some more configuration here in the future... probably use some kind of function that takes in all the args and spits back out a client.